airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -0,0 +1,226 @@
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: orders
+class Orders(IncrementalShopifyStream):
+    data_field = "orders"
+    cursor_field = "id"
+
+    def path(self, **kwargs) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: products
+class Products(IncrementalShopifyStream):
+    data_field = "products"
+    cursor_field = "id"
+
+    def path(self, **kwargs) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: abandoned_checkouts > checkouts
+class AbandonedCheckouts(IncrementalShopifyStream):
+    data_field = "checkouts"
+    cursor_field = "id"
+
+    def path(self, **kwargs) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: metafields
+class Metafields(IncrementalShopifyStream):
+    data_field = "metafields"
+    cursor_field = "id"
+
+    def path(self, **kwargs) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: custom_collections > custom collections
+class CustomCollections(IncrementalShopifyStream):
+    data_field = "custom_collections"
+    cursor_field = "id"
+
+    def path(self, **kwargs) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: collects
+class Collects(IncrementalShopifyStream):
+    data_field = "collects"
+    cursor_field = "id"
+
+    def path(self, **kwargs) -> str:
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/{self.data_field}.json"
+
+### STREAM: Refunds from orders
+class OrderRefunds(IncrementalShopifyStream):
+    data_field = "refunds"
+    cursor_field = "id"
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        order_id = stream_slice['order_id']
+        return f"https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/{self.api_version}/orders/{order_id}/{self.data_field}.json"
+
+    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+        orders_stream = Orders(self.shop, self.api_key, self.api_password, self.api_version)
+        for data in orders_stream.read_records(sync_mode=SyncMode.full_refresh):
+            yield from super().read_records(stream_slice={"order_id": data["id"]}, **kwargs)
+
+### STREAM: Refunds from orders
+class Transactions(IncrementalShopifyStream):
+    data_field = "transactions"
+    cursor_field = "id"
```

Comment:
if `id` is always the cursor field then we can probably just hardcode it at the incremental stream base class level. If a subclass needs to change it, then it can override the `cursor_field` property.